token/program-2022/src/instruction.rs

```diff
@@ -409,6 +431,148 @@ pub enum TokenInstruction {
         /// The freeze authority/multisignature of the mint.
         freeze_authority: COption<Pubkey>,
     },
+    /// Gets the required size of an account for the given mint as a little-endian
+    /// `u64`.
+    ///
+    /// Return data can be fetched using `sol_get_return_data` and deserializing
+    /// the return data as a little-endian `u64`.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[]` The mint to calculate for
+    GetAccountLength,
+    /// Initializes the empty mixin on a new account.
+    ///
+    /// The account must have a size 356, which requires it to use this mixin.
+    /// The size of 356 comes from the size of a multisig (355) plus 1 byte of
+    /// padding.
+    ///
+    /// Fails if the account has already been initialized, so must be called before
+    /// `InitializeAccount`. The account must have enough data allocated for all
+    /// mixins included in the mint.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]`  The account to initialize.
+    InitializeEmptyMixin,
```

Comment:
If we do it all in `InitializeAccount`, then yes no need for this

Reply:
and for mints, `InitializeMint` will have to do that check last then to sum up all the mixins and make sure the last byte is empty. I initially thought it would be best to write some data to that last byte, but you're right, it doesn't make a difference as long as the empty mixin is always last and implicit